facade/src/main/java/org/jboss/pnc/facade/providers/ProductMilestoneProviderImpl.java

```diff
@@ -253,7 +270,11 @@ public Page<MilestoneInfo> getMilestonesOfArtifact(String artifactId, int pageIn
             return new Page<>();
         }
 
-        CriteriaQuery<Tuple> query = milestoneInfoQuery(cb, milestoneIds);
+        Predicate<org.jboss.pnc.model.ProductMilestone> rsqlPredicate = rsqlPredicateProducer
+                .getCriteriaPredicate(milestoneInfoRSQLMapper, queryRsql);
```

Comment:
could the filtering be moved to getMilestoneIdByBuildRecord and getDependentMilestoneIds (or could it be all united to single query? But maybe keep it rather nice then optimized unless it's slow)

Reply:
LazyInitializationException